erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -1591,3 +1591,26 @@ def create_invoice_discounting(source_name, target_doc=None):
 def get_item_details(item_code):
 	item = frappe.get_doc("Item", item_code).as_dict()
 	return item
+
+@frappe.whitelist()
+def email_coas(docname):
+	sales_invoice = frappe.get_doc("Sales Invoice", docname)
+	attachments = []
+	if not sales_invoice.get("contact_email"):
+		frappe.throw(_("No contact email found"))
+	for item in sales_invoice.get('items'):
+		item.certificate_of_analysis = frappe.db.get_value("Batch", item.batch_no, "certificate_of_analysis")
+		coas = item.get("certificate_of_analysis")
+		if not coas:
+			frappe.msgprint(_("No Certificates of Analysis attached"))
+			return
```

Comment:
```suggestion
			continue
```
do you want to break the loop if even one of the items do not have a COA?

Reply:
as per the DOD the same functionality is done in Delivery Note, so i tried to follow the same procedure, as coas is not coming so they are breaking or returning the function null.
let me know if i need to change it to continue!